models/stock_picking.py

```diff
@@ -2,73 +2,80 @@
-        filtered_trailer_info = dict((k[2:], values[k]) for k in TRAILER_FIELDS if k in values)
+        # Filter values for transport info and substring the 'u_' from the keys to create correct transport field names
+        filtered_transport = dict((k[2:], values[k]) for k in TRANSPORT_FIELDS if k in values)
 
-        # Only create a trailer_info object if trailer_info data exists in values
-        if filtered_trailer_info:
+        # Only create a transport object if transport data exists in values
+        if filtered_transport:
             for record in self:
-                if not record.u_trailer_info_id:
+                if not record.u_transport_id:
                     # Add picking_id as this always exists
-                    filtered_trailer_info['picking_id'] = record.id
+                    filtered_transport['picking_id'] = record.id
 
-                    # Create the TrailerInfo object with created dict
-                    TrailerInfo.create(filtered_trailer_info)
+                    # Create the Transport object with created dict
+                    Transport.create(filtered_transport)
 
         return values
 
     @api.multi
     def write(self, values):
-        values = self._create_trailer_info_data(values)
+        values = self._create_transport_info_data(values)
         res = super(StockPicking, self).write(values)
         return res
 
     @api.model
     def create(self, values):
         res = super(StockPicking, self).create(values)
-        res._create_trailer_info_data(values)
+        res._create_transport_info_data(values)
         return res
 
     @api.multi
-    def _prepare_info(self, priorities=None, **kwargs):
-        data = super(StockPicking, self)._prepare_info(priorities=priorities, **kwargs)
-
-        # create a dictionary of the trailer data
-        trailer_data = dict((field, getattr(self, field)) for field in TRAILER_FIELDS if getattr(self, field, False))
-        # add it to the super data
-        data.update(trailer_data)
+    def _prepare_info(self, priorities=None, fields_to_fetch=None, **kwargs):
+        data = super(StockPicking, self)._prepare_info(priorities=priorities, fields_to_fetch=fields_to_fetch, **kwargs)
+
+        # If the stock picking requires transport
+        if self.u_requires_transport and (not fields_to_fetch or 'u_transport_id' in fields_to_fetch):
+            if self.u_transport_id:
+                # Get transport info and add it to data
+                data['u_transport_id'] = {'u_vehicle_sequence': self.u_vehicle_sequence,
+                                          'u_vehicle_description': self.u_vehicle_description,
+                                          'u_license_plate': self.u_license_plate,
+                                          'u_driver_name': self.u_driver_name}
+            else:
+                # for the case with no information, the front end expects an empty list
+                data['u_transport_id'] = []
```

Comment:
empty list or empty dictionary?

Reply:
After discussion, code was changed to only return fields that have been filled in